File: Creational/Builder/Python/ConcreteBuilder1.py
from AbstractBuilder import *
from ConcreteProduct1 import *

class ConcreteBuilder1(AbstractBuilder):

	def reset(self):
		self.product = ConcreteProduct1()
		return 
		
	def build_step_a(self):
		self.product.step_a_1()
		
	def build_step_b(self):
		self.product.step_b_1()
		
	def build_step_c(self):
		self.product.step_c_1()

File: Creational/Builder/C++/lib/ConcreteBuilder1.cpp
#include "../inc/ConcreteBuilder1.hpp"
#include "../inc/ConcreteProduct1.hpp"

ConcreteBuilder1::ConcreteBuilder1()
{

}

ConcreteBuilder1::~ConcreteBuilder1()
{

}

void ConcreteBuilder1::reset()
{
	m_product = std::make_shared<ConcreteProduct1>();
}

void ConcreteBuilder1::build_step_a()
{
	m_product->step_a_1();
}

void ConcreteBuilder1::build_step_b()
{
	m_product->step_b_1();
}

void ConcreteBuilder1::build_step_c()
{
	m_product->step_c_1();
}
